src/App.tsx

```diff
@@ -1,24 +1,312 @@
+      });
+  };
+
+  const onEditing = (todo: Todo | null) => {
+    setEditedTodo(todo);
+  };
+
+  const onEditSubmit = (
+    currentId: number,
+    event?: React.FormEvent<HTMLFormElement>,
+  ) => {
+    event?.preventDefault();
+
+    const currentTodo = todos.find(todo => todo.id === currentId) as Todo;
+
+    if (currentTodo.title === editedTodo?.title) {
+      setEditedTodo(null);
+
+      return;
+    }
+
+    if (!editedTodo?.title) {
+      onDeleteTodo(currentId);
+
+      return;
+    }
+
+    if (editedTodo) {
+      onUpdateTodos({
+        ...currentTodo,
+        title: editedTodo.title.trim(),
+      });
+    }
+  };
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          {todos.length > 0 && (
+            <button
+              type="button"
+              aria-label="toggle all"
+              className={classNames('todoapp__toggle-all', {
+                active: todos.length === completedTodos.length,
+              })}
+              data-cy="ToggleAllButton"
+              onClick={onToggleAll}
+            />
+          )}
+
+          <form onSubmit={onAddTodo}>
+            <input
+              disabled={submitDisabling}
+              value={title}
+              ref={inputRef}
+              data-cy="NewTodoField"
+              type="text"
+              className="todoapp__new-todo"
+              placeholder="What needs to be done?"
+              onChange={(event) => setTitle(event.target.value)}
+            />
+          </form>
+        </header>
+
+        <TodoList
+          todos={renderedTodos[filterBy]}
+          onToggleTodos={onToggleTodos}
+          tempTodo={tempTodo}
+          onDeleteTodo={onDeleteTodo}
+          deletedIds={deletedIds}
+          toggledTodo={toggledTodo}
+          editedTodo={editedTodo}
+          togglingAll={togglingAll}
+          isLoading={isLoading}
+          onEditing={onEditing}
+          onEditSubmit={onEditSubmit}
+        />
+
+        {/* Hide the footer if there are no todos */}
+        {todos.length > 0 && (
+          <footer className="todoapp__footer" data-cy="Footer">
+            <span className="todo-count" data-cy="TodosCounter">
+              {`${uncompletedTodos.length} items left`}
+            </span>
+
+            <TodosFilter
+              filters={filters}
+              filterBy={filterBy}
+              onFilterBy={setFilterBy}
+            />
+
+            <button
+              disabled={completedTodos.length === 0}
```

Comment:
```suggestion
              disabled={!completedTodos.length}
```

can be a bit more simplify